sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/SqlUpdate.java

```diff
@@ -88,11 +85,13 @@ public Handler buildHandler(Class<?> sqlObjectType, Method method) {
         }
 
         @Override
-        public Object invoke(Object target, Object[] args, HandleSupplier handle) {
-            String sql = handle.getConfig(SqlObjects.class).getSqlLocator().locate(sqlObjectType, getMethod());
-            Update update = handle.getHandle().createUpdate(sql);
-            applyCustomizers(update, args);
-            return this.returner.apply(update);
+        Update createStatement(Handle handle, String locatedSql) {
+            return handle.createUpdate(locatedSql);
+        }
+
+        @Override
+        void configureReturner(Update u, SqlObjectStatementConfiguration cfg) {
+            cfg.setReturner(() -> returner.apply(u));
```

Comment:
But if you capture it instead, you get it as an `Update` or similar.  Writing it as a Function that way introduces the statement into your scope twice as different static types.  So you either capture the correctly typed one anyway, or do an ugly cast in your lambda...

And why would we move the if/else chain into here from the constructor?  Not seeing the advantage